test/val/val_id_test.cpp

```diff
@@ -94,6 +94,52 @@ string sampledImageSetup = R"(
             %sampler_inst = OpLoad %sampler_type %s
 )";
 
+string BranchConditionalSetup = R"(
```

Comment:
FYI,  the style guide says "Variables of class type with static storage duration are forbidden"
For an explanation, see https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables 

I know this file breaks the rule multiple times.
Since this is test code, it's not critical to fix.  But we do enforce the rule in the library code.

The alternative is either  const char array, pointer-to-const-char, or a function returning a static string value.

Reply:
That rule exists probably because a very large program would initialize all such variables on startup even if it will never use them. Doesn't really apply in this case.

Reply:
It's for a bunch of reasons, but the most critical ones are the confusion and bugs that arise from indeterminate initialization order and indeterminate destruction order. 

It happens in real life with SPIRV-Tools:   When SPIRV-Tools is used in the MicrosoftDirectXShaderCompiler, the main process replaces the memory allocator at some point (I believe after 'main' starts), and if you have variables of the banned type, then there is trouble at process shutdown because the deallocator mismatches the allocator.

Yes, that does not apply for test code (which this is an instance of).  That's why it's a FYI in this case.